qrenderdoc/Windows/MainWindow.cpp

```diff
@@ -1408,8 +1408,9 @@ void MainWindow::on_action_Attach_to_Running_Instance_triggered()
 
 void MainWindow::on_action_Manage_Remote_Servers_triggered()
 {
-  // the manager deletes itself when all lookups terminate
-  RDDialog::show(new RemoteManager(m_Ctx, this));
+  RemoteManager *dialog = new RemoteManager(m_Ctx, this);
+  RDDialog::show(dialog);
+  delete dialog;
```

Comment:
This is going to crash. As the comment you removed implies, the lookups that happen to remote hosts asynchronously callback into the RemoteManager, so it can't be deleted externally behind their back. It needs to keep itself alive until all of those succeed, without blocking on those requests.

What's the crash you ran into specifically?